packages/muelu/test/unit_tests/Hierarchy.cpp

```diff
@@ -1592,11 +1592,13 @@ namespace MueLuTests {
     // Write matrices out, read fine A back in, and check that the read was ok
     // by using a matvec with a random vector.
     char t[] = "XXXXXX";
+    int fd;
     if (comm->getRank() == 0)
-      mkstemp(t); //mkstemp() creates a temporary file. We use the name of that file as
-                  //the suffix for the various data files produced by Hierarchy::Write().
-                  //A better solution would be to write to a file stream, but this would
-                  //involve writing new interfaces to Epetra's file I/O capabilities.
+      fd = mkstemp(t); //mkstemp() creates a temporary file. We use the name of that file as
+                       //the suffix for the various data files produced by Hierarchy::Write().
+                       //A better solution would be to write to a file stream, but this would
+                       //involve writing new interfaces to Epetra's file I/O capabilities.
+    TEST_INEQUALITY(fd, -1); // if the file descriptor is -1, it failed
```

Comment:
`fd` is uninitialized on procs other than rank 0. Might this affect this test? Maybe initialize `fd` above to be on the safe side...

Reply:
Good point. It might introduce random unspecified behavior, which is the last thing we need in the testsuite... especially since the only failed test from the AT was a random failure